### PR TITLE
Fixes lighting hard deleting

### LINKED ISSUE #30589: /atom/movable/lighting_object is stupid expensive to hard delete and is causing alot of lag.
````
/atom/movable/lighting_object
    Total Hard Deletes 556
    Time Spent Hard Deleting: 313247ms
````

@PJB3005 any ideas? @tgstation/issue-managers high priority please

### PR DESCRIPTION
**Pull request by MrsTonedOne**
:cl: Fun Police
del: Removed some of the free lag. 
/:cl:

lighting was getting thrown by transit turfs

fixes #30589

### CODE REVIEW

File: code/__DEFINES/misc.dm [hunk 1]
```diff
@@ -356,7 +356,6 @@ GLOBAL_LIST_INIT(ghost_others_options, list(GHOST_OTHERS_SIMPLE, GHOST_OTHERS_DE
 #define debug_admins(msg) if (GLOB.Debug2) to_chat(GLOB.admins, "DEBUG: [msg]")
 #define debug_world_log(msg) if (GLOB.Debug2) log_world("DEBUG: [msg]")
 
-#define COORD(A) "([A.x],[A.y],[A.z])"
```
Comment (MrStonedOne) [inline code/__DEFINES/misc.dm]: This was a dupe and the other one was better

File: code/modules/lighting/lighting_object.dm [hunk 2]
```diff
@@ -33,19 +36,28 @@
...
 
 		return ..()
+
 	else
 		return QDEL_HINT_LETMELIVE
 
 /atom/movable/lighting_object/proc/update()
-	var/turf/T = loc
-	if (!istype(T)) // Erm...
+	loc
```
Comment (RemieRichards) [inline code/modules/lighting/lighting_object.dm]: excuse me

File: code/modules/lighting/lighting_object.dm [hunk 3]
```diff
@@ -33,19 +36,28 @@
...
+	if (loc != myturf)
 		if (loc)
+			var/turf/oldturf = get_turf(myturf)
+			var/turf/newturf = get_turf(loc)
```
Comment (RemieRichards) [inline code/modules/lighting/lighting_object.dm]: you gonna use this var buddy?

**Comment by MrStonedOne** [comment]
>that feel when you forget to push the updated version

**Comment by RemieRichards** [comment]
Shhh, it's ok, no-one saw.

**Comment by RemieRichards** [comment]
>lighting was getting thrown by transit turfs

BUT ITS CALLED FASTER THAN LIGHT TRAVEL!! from our relative view the light *is* getting thrown!!!

**Comment by Arianya** [comment]
Thank god

**Comment by kevinz000** [comment]
lmao

File: code/game/turfs/space/transit.dm [hunk 4]
```diff
@@ -64,8 +64,8 @@
 			_y = min
 
 	var/turf/T = locate(_x, _y, _z)
-	AM.loc = T
-	AM.newtonian_move(dir)
```
Comment (Cyberboss) [inline code/game/turfs/space/transit.dm]: will people still drift without this?

File: code/modules/lighting/lighting_object.dm [hunk 5]
```diff
@@ -32,24 +35,26 @@
...
-		if (istype(T))
-			T.lighting_object = null
-			T.luminosity = 1
+		if (loc != myturf)
+			var/turf/oldturf = get_turf(myturf)
+			var/turf/newturf = get_turf(loc)
+			stack_trace("A lighting object was qdeleted with a different loc then it is suppose to have ([COORD(oldturf)] -> [COORD(newturf)])")
+		if (isturf(myturf))
+			myturf.lighting_object = null
+			myturf.luminosity = 1
```
Comment (Cyberboss) [inline code/modules/lighting/lighting_object.dm]: TRUE

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/lighting/lighting_object.dm [hunk 6]
```diff
@@ -14,14 +14,17 @@
...
 	verbs.Cut()
 
-	var/turf/T         = loc // If this runtimes atleast we'll know what's creating overlays in things that aren't turfs.
-	T.lighting_object = src
-	T.luminosity       = 0
+	myturf = loc
+	if (myturf.lighting_object)
+		qdel(myturf.lighting_object, force = TRUE)
+	myturf.lighting_object = src
+	myturf.luminosity = 0
```
[the remark this answers is not in the document]
Comment (MrStonedOne) [reply code/modules/lighting/lighting_object.dm]: luminosity is a int, not a bool,

File: code/modules/lighting/lighting_object.dm (hunk 5 above)
Comment (MrStonedOne) [reply code/modules/lighting/lighting_object.dm]: luminosity is a int, not a bool,

File: code/game/turfs/space/transit.dm (hunk 4 above)
Comment (MrStonedOne) [reply code/game/turfs/space/transit.dm]: Yes, forced move calls Moved and that calls this

**Pull request merged**